Makefile

```diff
@@ -1,30 +1,30 @@
-# CFLAGS = -Xcc -DMONGOC_COMPILATION -Xcc -DBSON_COMPILATION -Xcc -ISources/libbson -Xcc -ISources/libbson/include -Xcc -ISources/libbson/generated -Xcc -ISources/libmongoc/generated
-# LDFLAGS = -Xlinker -lsasl2 -Xlinker -lz
-CFLAGS =
-LDFLAGS =
-
-# If FILTER is not provided, a default filter of `MongoSwiftTests` will be used.
-# Else, any test matching the filter in *either target* (MongoSwiftTests or MongoSwiftBenchmarks) will run.
+# if provided, FILTER is used as the --filter argument to `swift test`. 
 ifdef FILTER
 	FILTERARG = --filter $(FILTER)
 else
-	FILTERARG = --filter MongoSwiftTests
+	FILTERARG =
 endif
 
+define check_for_gem
+	gem list $(1) -i > /dev/null || gem install $(1) || { echo "ERROR: Failed to locate or install the ruby gem $(1); please install yourself with 'gem install $(1)' (you may need to use sudo)"; exit 1; }
+endef
+
 all:
-	swift build -v $(CFLAGS) $(LDFLAGS)
+	swift build -v
```

Comment:
So ` $(CFLAGS) $(LDFLAGS)` is removed here, but looking at the diff, it seems like they never did anything in the first place?:

```make
# CFLAGS = -Xcc -DMONGOC_COMPILATION -Xcc -DBSON_COMPILATION -Xcc -ISources/libbson -Xcc -ISources/libbson/include -Xcc -ISources/libbson/generated -Xcc -ISources/libmongoc/generated	# if provided, FILTER is used as the --filter argument to `swift test`. 
# LDFLAGS = -Xlinker -lsasl2 -Xlinker -lz	
CFLAGS =	
LDFLAGS =
```
Seems like they were both empty, and the substantial assignments above were commented out.

So was this just some clean-up you happened to do and not actually related to the PR?

It's also worth considering that an empty variable is not useless in Makefiles and gives degree of control to the user. Is it possible then that we had empty `CFLAGS` and `LDFLAGS` and users could run something like `make CFLAGS="..."` or similar? They still can do this,  even without the variables in the Makefile, but I _think_ it is better practice to keep such variables as empty in the Makefile for the sake of documentation.

Reply:
Sounds good to me.

It's odd that it gets replaced because this is not the behavior I've seen in other Makefiles (where an empty target doesn't overwrite the user-supplied version). I was able to get a minimal Makefile working with an empty variable assignment that still took on the user-supplied value on my Macbook just now... Perhaps there is something more subtle going on here... 🤔 

But in either case, sounds good to me. 👍

Reply:
lol well you know what they say about assuming... just checked, you were right, I find this behavior very unintuitive though 🤔 I guess these are more of default values, whereas I was reading them as variable assignments. 
in any case, ya, I don't think we need them anymore, if we change our minds we can come back and remember what this looked like